docs/tutorial.rst

```diff
@@ -72,3 +78,49 @@ for this machine learning model.
     label_name = sess.get_outputs()[0].name
     
     pred_onx = sess.run([label_name], {input_name: X_test.astype(numpy.float32)})[0]
+
+.. index:: initial types
+
+convet_sklearn, to_onnx, initial_types
+======================================
+
+The module implements two functions:
+:func:`convert_sklearn <skl2onnx.convert_sklearn>` and
+:func:`to_onnx <skl2onnx.to_onnx>`. The first one
+was used in the previous examples, it requires two
+mandatory arguments:
+
+* a *scikit-learn* model or a pipeline
+* initial types
+
+*scikit-learn* does not store information about
+the training dataset. It is not always possible to retrieve
+the number of features or their types. That's why the
+function needs another argument called *initial_types*.
+In many cases, the training datasets is a numerical matrix
+*X_train*. Then it becomes
+``initial_type=[('X', FloatTensorType([None, X_train.shape[1]]))]``.
+*X* is the name of this unique input, the second term indicates the
+type and shape. The shape is ``[None, X_train.shape[1]]``,
+the first dimension is the number of rows, undefined, and the 
```

Comment:
Shouldn't we say `variable` instead of `undefined`?

Reply:
I proposed a new formulation.